### PR TITLE
[Dashboard] Fix Ray Dashboard command error messages

### PR DESCRIPTION
**Pull request by mfitton**
## Why are these changes needed?
This fixes a bug in the `ray dashboard` command. It used to spin and try to forward new ports because of an error in the command being run. The issue is that the command to sleep forever was quoted. Because it was quoted, Linux attempted to interpret it as a single command, as opposed to running bash evaluation on the line as was desired. In addition, I made the error log message contain more information and be friendlier, and I removed the behavior of continuing to choose a random port until hitting one that you want now that both the local and remote ports can be specified.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by wuisawesome** [comment]
iirc the retry loop was originally intended to allow us to port forward multiple clusters to local host at the same time (i.e. have a dashboard for a dev cluster and another for a big remote cluster). I don't feel very strongly about that behavior, but was just curious if there was a reason for removing it

**Review by wuisawesome — APPROVED** [review]

**Comment by edoakes** [comment]
@mfitton I think there were issues with CI in master, can you merge latest and push to trigger CI?

**Comment by mfitton** [comment]
I encountered a user that was annoyed that even though he specified a port, it still kept trying to use other ports for the forwarding. That said, I could have it still do the counting-up behavior when the port being used is the default port.

**Comment by wuisawesome** [comment]
> I encountered a user that was annoyed that even though he specified a port, it still kept trying to use other ports for the forwarding. That said, I could have it still do the counting-up behavior when the port being used is the default port.

Personally, I think we should handle this as a separate feature (either a `--remote-port`/`--local-port` flag or a `ray port-forward --block` command. It's probably already achievable via `ray exec -L <port> while true; "do sleep(10000); done;"` but conflating port forwarding and execution is in many ways a leaky abstraction around ssh (`kubectl` doesn't support this for example)

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -129,34 +129,28 @@ def dashboard(cluster_config_file, cluster_name, port, remote_port):
...
+                   " remote port {}".format(port, remote_port))
+        # We want to probe with a no-op that returns quickly to avoid
+        # exceptions caused by network errors.
+        exec_cluster(
+            cluster_config_file,
+            override_cluster_name=cluster_name,
+            port_forward=port_forward)
+        click.echo("Successfully established connection.")
+    except Exception as e:
+        click.echo(
```
Comment (richardliaw) [inline python/ray/scripts/scripts.py:148]: tip: raise click.ClickException(...) from None

to achieve the same thing

**Pull request merged**